### ISSUE TITLE
GauXC::XCIntegrator<MatrixType>::eval_exc_vxc gives different results with same input 

### ISSUE THREAD

**Issue by samslattery** [wavefunction91/GauXC#8]
When eval_exc_vxc is called for object of type GauXC::XCIntegrator<MatrixType> multiple times (with same input density) different energy values are produced. For example, simply adding the below to file `tests/xc_integrator.cxx` will demonstrate the issue.
```
  auto [ EXC2, VXC2 ] = integrator.eval_exc_vxc( P );
  std::cout << "EXC2 = " << EXC2 << std::endl;
  auto [ EXC3, VXC3 ] = integrator.eval_exc_vxc( P );
  std::cout << "EXC3 = " << EXC3 << std::endl;
```
Output I got 
```
EXC2 = -24.4746
EXC3 = -23.6536
```

**Comment by wavefunction91** [comment]
Definitely not the expected behaviour, thanks for pointing it out!

**Comment by wavefunction91** [comment]
@samslattery Can you confirm [bug-fix/integrator_state_persistence](https://github.com/wavefunction91/GauXC/tree/bug-fix/integrator_state_persistence) / #9  fixes your issue?

**Comment by samslattery** [comment]
Yes, this fixed my issue. Thanks!
